esp/src/eclwatch/PreflightDetailsWidget.js

```diff
@@ -0,0 +1,247 @@
+#    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+############################################################################## */
+define([
+    "dojo/_base/declare",
+    "dojo/_base/lang",
+    "dojo/i18n",
+    "dojo/i18n!./nls/hpcc",
+    "dojo/_base/array",
+
+    "dijit/registry",
+    "dijit/form/Button",
+    "dijit/ToolbarSeparator",
+    "dijit/Dialog",
+    "dijit/form/TextBox",
+
+    "hpcc/GridDetailsWidget",
+    "hpcc/ESPUtil",
+
+    "dgrid/selector",
+
+], function (declare, lang, i18n, nlsHPCC, arrayUtil,
+                registry, Button, ToolbarSeparator, Dialog, TextBox,
+                GridDetailsWidget, ESPUtil,
+                selector) {
+    return declare("PreflightDetailsWidget", [GridDetailsWidget], {
+        i18n: nlsHPCC,
+
+        preflightTab: null,
+        preflightWidgetLoaded: false,
+        gridTitle: nlsHPCC.title_PreflightResults,
+        idProperty: "__hpcc_id",
+
+        init: function (params, route) {
+            if (this.inherited(arguments))
+                return;
+            this.initalized = false;
+            this.params = params;
+            this.getColumns(params);
+            this.refresh(params, route);
+        },
+
+        refresh: function (params, route) {
+            route === "machines" ? this.refreshMachinesGrid(params) : this.refreshClusterGrid(params);
+        },
+
+        createGrid: function (domID) {
+            var context = this;
+
+            this.openButton = registry.byId(this.id + "Open");
+            this.openButton.set("disabled", true);
+
+            var retVal = new declare([ESPUtil.Grid(true, true)])({
+                store: this.store,
+                columns: context.getColumns(),
+            }, domID);
+            return retVal;
+        },
+
+        getColumns: function (params) {
+            var context = this;
+            var dynamicColumns = {
+                Location: {label: this.i18n.Location},
+                Component: {label: this.i18n.Component},
+                Condition: {label: this.i18n.Condition},
+                State: {label: this.i18n.State}
+            }
+
+            if (params) {
+                var finalColumns = params.Columns.Item;
+                for (index in finalColumns) {
+                    var clean = finalColumns[index].replace(/([~!@#$%^&*()_+=`{}\[\]\|\\:;'<>,.\/? ])+/g, '').replace(/^(-)+|(-)+$/g,'');
```

Comment:
Since the response returns "Column Name" I am removing any spaces and or special characters in it so that they come back like so: "ColumnName"